components/common/src/templating/hooks.rs

```diff
@@ -440,54 +440,81 @@ pub struct HookOutput<'a> {
 
 impl<'a> HookOutput<'a> {
     fn new(stdout_log: &'a Path, stderr_log: &'a Path) -> Self {
-        HookOutput { stdout_log_file: stdout_log,
-                     stderr_log_file: stderr_log, }
+        Self { stdout_log_file: stdout_log,
+               stderr_log_file: stderr_log, }
+    }
+
+    pub fn stdout(&self) -> Result<BufReader<File>> {
+        Ok(BufReader::new(File::open(&self.stdout_log_file)?))
     }
 
-    pub fn stdout(&self) -> Option<BufReader<File>> {
-        match File::open(&self.stdout_log_file) {
-            Ok(f) => Some(BufReader::new(f)),
-            Err(_) => None,
+    pub fn stdout_str(&self) -> Option<String> {
+        let result = self.stdout_str_impl();
+        if let Err(e) = &result {
+            error!("Failed to read {:?}, {}", self.stdout_log_file, e);
         }
+        result.ok()
+    }
+
+    pub fn stderr(&self) -> Result<BufReader<File>> {
+        Ok(BufReader::new(File::open(&self.stderr_log_file)?))
     }
```

Comment:
Yeah, it is not ideal. I think `stdout` and `stdout_str` are both useful. We could make users of `HookOutput` manually convert to a `String`, but I think having the helper is useful especially when we add [limit to the length](https://github.com/habitat-sh/habitat/pull/6905#discussion_r319606085). The `*_impl`s are just needed because it lets us use the `?` operator which I think is worth the negative of having another method. So I guess I don't see an obvious way to simplify this, but am certainly open to suggestions.